arch/arm/core/irq_manage.c

```diff
@@ -84,19 +84,18 @@ int _arch_irq_is_enabled(unsigned int irq)
  */
 void _irq_priority_set(unsigned int irq, unsigned int prio, u32_t flags)
 {
-	/* Hardware priority levels 0 and 1 reserved for Kernel use.
-	 * So we add 2 to the requested priority level. If we support
-	 * ZLI, 2 is also reserved so we add 3.
+	/* Hardware priority levels can be reserved for Kernel use.
+	 * So we add to the requested priority level.
```

Comment:
This line is incomplete; please, specify what exactly is added.

Reply:
I think we need to be explicit here. The deleted text is definitely outdated, buy we should not replace it with unclear text. IMO
- the comment is wrong for ARMv6-M, ARMv8-M Baseline: we do not reserve anything there
- For ARMv7-M, ARMv8-M Mainline Kernel reserves level-0 for faults, **and**, level-1 for zero-latency IRQs (if configured to do so)
- all the above are reflected to `_IRQ_PRIO_OFFSET` which is, thus, _added_ to the given _prio_ for normal (i.e. not Zero-latency) IRQs

I think the comment should contain all this information.

Reply:
The explicit text is what caused it to be outdated in the first place.
There is no need to tell which IRQs are reserved for what at this point, we only need to explain why we are offsetting them.

Reply:
Well, before we were just "telling that Kernel uses 0 and 1" which was hardly possible to maintain, and was not educating the reader of the code. I think if we refer to the explicit reasons we add to the maintainability/educability  of the code.

However, no strong opinion. If you want to just explain in detail why we perform the offset, this is OK with me.